libparsec/crates/platform_mountpoint/src/unix/mod.rs

```diff
@@ -0,0 +1,438 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+mod inode;
+
+use fuser::{
+    FileAttr, FileType, Filesystem, MountOption, ReplyAttr, ReplyCreate, ReplyData, ReplyDirectory,
+    ReplyEmpty, ReplyEntry, ReplyOpen, ReplyWrite, Request, Session,
+};
+use libc::{EINVAL, ENAMETOOLONG, ENOENT, ENOTEMPTY, EPERM};
+use libparsec_types::{anyhow, FileDescriptor};
+use std::{ffi::OsStr, path::Path, time::Duration};
+
+use crate::{
+    error::MountpointError, EntryInfo, EntryInfoType, FileSystemMounted, FileSystemWrapper,
+    MountpointInterface, WriteMode,
+};
+
+use inode::Inode;
+pub(crate) use inode::InodeManager;
+
+/// TODO: Do we need to change this value ?
+/// Validity timeout
+const TTL: Duration = Duration::ZERO;
+
+/// TODO: Do we need to handle any other GENERATION ?
+///
+/// If the file system will be exported over NFS, the inode/generation pairs need
+/// to be unique over the file system's lifetime (rather than just the mount
+/// time). So if the file system reuses an inode after it has been deleted, it
+/// must assign a new, previously unused generation number to the inode at the
+/// same time.
+const GENERATION: u64 = 0;
+const BLOCK_SIZE: u64 = 512;
+
+impl From<EntryInfoType> for FileType {
+    fn from(value: EntryInfoType) -> Self {
+        match value {
+            EntryInfoType::Dir => Self::Directory,
+            EntryInfoType::File => Self::RegularFile,
+            EntryInfoType::ParsecEntryInfo => Self::RegularFile,
+        }
+    }
+}
+
+fn entry_info_to_file_attr(entry_info: EntryInfo, inode: Inode) -> FileAttr {
+    let created = entry_info.created;
+    let updated = entry_info.updated;
+
+    let (kind, perm) = match entry_info.ty {
+        EntryInfoType::Dir => (FileType::Directory, 0o755),
+        EntryInfoType::File => (FileType::RegularFile, 0o644),
+        EntryInfoType::ParsecEntryInfo => (FileType::RegularFile, 0),
+    };
+
+    FileAttr {
+        ino: inode.into(),
+        size: entry_info.size,
+        blocks: (entry_info.size + BLOCK_SIZE - 1) / BLOCK_SIZE,
+        atime: updated.into(),
+        mtime: updated.into(),
+        ctime: updated.into(),
+        crtime: created.into(),
+        kind,
+        perm,
+        nlink: 0,
+        uid: entry_info.id.as_u128() as u32,
+        gid: entry_info.id.as_u128() as u32,
+        rdev: 0,
+        blksize: BLOCK_SIZE as u32,
+        flags: 0,
+    }
+}
+
+impl<T: MountpointInterface> Filesystem for FileSystemWrapper<T> {
+    fn lookup(&mut self, _req: &Request, parent: u64, name: &OsStr, reply: ReplyEntry) {
```

Comment:
add a comment to explain lookup / forget and inode/path relationship

Reply:
Done